usecases/cluster/state.go

```diff
@@ -208,9 +208,16 @@ func (s *State) NodeHostname(nodeName string) (string, bool) {
 func (s *State) NodeAddress(id string) string {
 	s.listLock.RLock()
 	defer s.listLock.RUnlock()
-	for _, mem := range s.list.Members() {
-		if mem.Name == id {
-			return mem.Addr.String()
+
+	// create copy of list to avoid data race
```

Comment:
I'm not entirely sure but It seems that instead of one copy, we're now making N copies and returning the address of one of them

Reply:
N copies of the nodes `s.list.Members()` will return `[]*Node`, so to avoid accessing the address, just copy the node info

Reply:
We might need to consider that these are only shallow copies

Reply:
This seems to me like a bug in the member list implementation, failing to provide a safe way to query the node address. Can you reliably reproduce the issue with race detection?

Reply:
it's bug, the point is the exposed `*Node` gets updated while it's read, to solve the problem `node.Addr()`  would need to have the node Mutex to protect it , but that will need sometime to report the bug and open PR and resolve. what i have thought about is just copying the node for now. 

to properly testing it that could require time because we need to mock all the network changes and also add proper unit tests to the state in our unit etst but at the moment we don't have any tests for that

Reply:
It is possible to reproduce the issue with the Go race detector?

Reply:
we don't have tests :) , i can spend time to test for sure it's a bout what is perio atm!

Reply:
It is not clear to me how we can validate that this PR fixes the issue. That's why I was asking if it is possible to run the race detector to reproduce this issue